dspMetadataGUI/util/converter.py

```diff
@@ -0,0 +1,763 @@
+        return res
+    except Exception:
+        return f'XX: CC URL: {url}'
+
+
+def _get_geonames_name(url: str):
+    """Get display text for a geonames URL"""
+    if url.endswith('/'):
+        url = url.removesuffix('/')
+    try:
+        if re.search('\/countries\/', url):
+            r = requests.get(url)
+            soup = BeautifulSoup(r.content, 'html.parser')
+            tables = soup.select('table')
+            table = tables[1]
+            link = table.select_one('a')
+            url = link['href']
+        if url.endswith(".html"):
+            url = url.rsplit("/", 1)[0]
+        gn_id = url.rsplit("/")[-1]
+        base = f'http://api.geonames.org/getJSON'
+        payload = {'geonameId': gn_id,
+                   'username': 'blandolt'}
+        r = requests.get(base, params=payload)
+        resp = r.json()
+        name = resp.get('name')
+        if not name:
+            raise Exception()
+        return name
+    except Exception:
+        return f'XX: Geonames URL: {url}'
+
+
+def _get_url_type(propID):
+    """Get URL type from schema.propertyID"""
+    if propID.startswith("SKOS"):
+        return "Skos"
+    if propID.startswith("Geonames"):
+        return "Geonames"
+    if propID.startswith("Pleiades"):
+        return "Pleiades"
+    if propID.startswith("ORCID"):
+        return "ORCID"
+    if propID.startswith("Periodo"):
+        return "Periodo"
+    if propID.startswith("ChronOntology") or propID.startswith("dainst."):
+        return "Chronontology"
+    if propID.startswith("GND"):
+        return "GND"
+    if propID.startswith("VIAF"):
+        return "VIAF"
+    if propID.startswith("Creative Commons"):
+        return "Creative Commons"
+    else:
+        return "URL"
+
+
+def validate(data, verbose=False):
+    """Validates JSON
+
+    Validates json serialized data against the schema of the API specification.  
+
+    [Schema](https://raw.githubusercontent.com/dasch-swiss/dasch-service-platform/main/docs/services/metadata/schema-metadata.json)
+
+    Args:
+        data ([type]): [description]
+    """
+    r = requests.get(schema_url)
+    schema = r.json()
+    try:
+        if verbose:
+            print("Validating...")
+        validator = jsonschema.Draft7Validator(schema)
+        valid = True
+        for e in validator.iter_errors(data):
+            if verbose:
+                print(f'Validation Error: {e.message}')
+            valid = False
+        if valid and verbose:
+            print("JSON is valid.")
+        return valid
+    except jsonschema.ValidationError as val:
+        if verbose:
+            print(val.message)
+        return False
+
+
+if __name__ == "__main__":
+    # files = ['maximal.ttl',
+    #          'rosetta.ttl',
+    #          'limc.ttl',
+    #          'awg.ttl',
+    #          'hdm.ttl'
+    #          ]
+    files = ['drawings-gods.ttl', 'awg.ttl']
+    results = {}
+    if not os.path.exists('out'):
+        os.mkdir('out')
```

Comment:
Gemäss https://stackoverflow.com/questions/273192/how-can-i-safely-create-a-nested-directory könnte es hier zu einer Race Condition kommen. Alternativ könnte man folgenden Code benutzen:

`os.makedirs("path/to/directory", exist_ok=True)`

Oder:

```
from pathlib import Path
Path("/my/directory").mkdir(parents=True, exist_ok=True)
```

Reply:
Interesting! I wasn't aware of that issue, thanks for pointing it out!